src/Promitor.Core.Scraping/Configuration/Model/Metrics/MetricDefinition.cs

```diff
@@ -4,6 +4,26 @@ namespace Promitor.Core.Scraping.Configuration.Model.Metrics
 {
     public abstract class MetricDefinition
     {
+        protected MetricDefinition()
+        {
+        }
+
+        protected MetricDefinition(
+            AzureMetricConfiguration azureMetricConfiguration,
+            string description,
+            string name,
+            string resourceGroupName,
+            Dictionary<string, string> labels,
+            Scraping scraping)
+        {
+            AzureMetricConfiguration = azureMetricConfiguration;
```

Comment:
I don't think we want to create invalid metric definitions so suggest to guard for invalid values. Thoughts?

Reply:
You just thinking of adding `Guard.NotNull()` calls here? If so makes perfect sense to me.

Reply:
I think that should be good enough for now, for strings you should be able to use NotNullOrEmpty or something similar

Reply:
Resolving this conversation as it's no longer applicable.